staging/src/k8s.io/client-go/rest/exec.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rest
+
+import (
+	"fmt"
+
+	"k8s.io/client-go/pkg/apis/clientauthentication"
+	clientauthenticationapi "k8s.io/client-go/pkg/apis/clientauthentication"
+)
+
+// This file contains Config logic related to exec credential plugins.
+
+// ConfigToExecCluster creates a clientauthenticationapi.Cluster with the corresponding fields from
+// the provided Config. If config.ExecProvider is nil or config.ExecProvider.ProvideClusterInfo is
+// false, then this function will return an empty clientauthenticationapi.Cluster.
+func ConfigToExecCluster(config *Config) (*clientauthenticationapi.Cluster, error) {
+	if config.ExecProvider == nil || !config.ExecProvider.ProvideClusterInfo {
+		return &clientauthentication.Cluster{}, nil
```

Comment:
this is inconsistent with the API doc that indicates the cluster field will be nil if ProvideClusterInfo is false. It looks like we only call this in one non-test location; should we put the burden of doing this check on the caller instead and assume if we are called that `config.ExecProvider != nil && config.ExecProvider.ProvideClusterInfo == true`?

Reply:
09323e5c557

This will likely change once we figure out where to put this thing...